monitor_event_handler.go

```diff
@@ -47,9 +103,100 @@ func (h *MonitorEventHandler) OnConnect(_ *centrifuge.Client, e centrifuge.Conne
 			}
 		}()
 	}
+
+	h.logger.Info(h.ctx, "[MONITOR] PROCESS BLOCK HEADERS")
+	err := h.ProcessBlockHeaders(h.ctx, client)
+	if err != nil {
+		h.logger.Error(h.ctx, fmt.Sprintf("[MONITOR] ERROR processing block headers: %s", err.Error()))
+	}
+
+	h.logger.Info(h.ctx, "[MONITOR] PROCESS BLOCKS")
+	go func() {
+		err := h.ProcessBlocks(h.ctx, client)
+		if err != nil {
+			h.logger.Error(h.ctx, fmt.Sprintf("[MONITOR] ERROR processing blocks: %s", err.Error()))
+		}
+	}()
+
 	h.monitor.Connected()
 }
 
+// ProcessBlocks processes all transactions in blocks that have not yet been synced
+func (h *MonitorEventHandler) ProcessBlocks(ctx context.Context, client *centrifuge.Client) error {
+
+	for {
+		// get all block headers that have not been marked as synced
+		blockHeaders, err := h.buxClient.GetUnsyncedBlockHeaders(ctx)
+		if err != nil {
+			h.logger.Error(h.ctx, err.Error())
+		} else {
+			h.logger.Info(ctx, fmt.Sprintf("[MONITOR] processing block headers: %d", len(blockHeaders)))
+			for _, blockHeader := range blockHeaders {
+				h.logger.Info(ctx, fmt.Sprintf("[MONITOR] Processing block %d: %s", blockHeader.Height, blockHeader.ID))
+				handler := &blockSubscriptionHandler{
+					debug:     h.debug,
+					logger:    h.logger,
+					monitor:   h.monitor,
+					buxClient: h.buxClient,
+					ctx:       context.Background(),
+				}
+				handler.wg.Add(1)
+
+				var subscription *centrifuge.Subscription
+				subscription, err = client.NewSubscription("block:sync:" + blockHeader.ID)
+				if err != nil {
+					h.logger.Error(h.ctx, err.Error())
+				} else {
+					h.logger.Info(ctx, fmt.Sprintf("[MONITOR] Starting block subscription: %v", subscription))
+					subscription.OnPublish(handler)
+					subscription.OnUnsubscribe(handler)
+					err = subscription.Subscribe()
+					if err != nil {
+						h.logger.Error(h.ctx, err.Error())
+					} else {
+						h.logger.Info(ctx, "[MONITOR] Waiting for waitgroup to finish")
+						handler.wg.Wait()
+
+						// save that block header has been synced
+						blockHeader.Synced.Valid = true
+						blockHeader.Synced.Time = time.Now()
+						err = blockHeader.Save(ctx)
+						if err != nil {
+							h.logger.Error(h.ctx, err.Error())
+						}
+					}
+				}
+			}
+		}
+
+		time.Sleep(30 * time.Second)
```

Comment:
Wait 30 seconds? This should be a constant (in definitions or config)

Reply:
Anytime we have values like this, IMO they should live as constant or an option, but all in one place to make it easy to find and configure